src/config/mod.rs

```diff
@@ -43,6 +43,7 @@ pub fn parse_configs(
     let mut listen_ips_4: HashMap<String, Vec<Ipv4Addr>> = HashMap::new();
     let mut listen_ips_6: HashMap<String, Vec<Ipv6Addr>> = HashMap::new();
     let mut ctr_dns_server: HashMap<IpAddr, Option<Vec<IpAddr>>> = HashMap::new();
+    let mut network_dns_server: HashMap<String, Option<Vec<IpAddr>>> = HashMap::new();
```

Comment:
seems kinda point less to have an extra level of Option<>, I would just use `HashMap<String, Vec<IpAddr>>`.
You are already only adding the key if you have servers so the None value is not used.